desc/compute/_equil.py

```diff
@@ -85,6 +85,196 @@ def _J_sup_theta(params, transforms, profiles, data, **kwargs):
     return data
 
 
+@register_compute_fun(
+    name="J^theta_PEST",
+    label="J^{\\theta_{PEST}}",
+    units="A \\cdot m^{-3}",
+    units_long="Amperes / cubic meter",
+    description="Contravariant PEST poloidal component of plasma current density",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=["J", "e^theta_PEST"],
+    aliases=["J^vartheta"],
+)
+def _J_sup_theta_PEST(params, transforms, profiles, data, **kwargs):
+    data["J^theta_PEST"] = dot(data["J"], data["e^theta_PEST"])
+    return data
+
+
+# TODO: Generalize for a general zeta before #568
+@register_compute_fun(
+    name="(J^theta_PEST_v)|PEST",
+    label="\\partial_{\\vartheta} J^{\\theta_{PEST}}",
+    units="A \\cdot m^{-3}",
+    units_long="Amperes / cubic meter",
+    description="Contravariant PEST poloidal component of plasma current density"
+    + " derivative w.r.t poloidal PEST coordinate",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=[
+        "J^theta_t",
+        "J^theta",
+        "J^zeta",
+        "J^zeta_t",
+        "theta_PEST_t",
+        "theta_PEST_z",
+        "theta_PEST_tz",
+        "theta_PEST_tt",
+    ],
+    aliases=["J^vartheta_v|PEST"],
+)
+def _J_sup_theta_PEST_v_PEST(params, transforms, profiles, data, **kwargs):
+    # J^ϑ = J^θ (∂ϑ/∂θ) + J^ζ (∂ϑ/∂ζ)
```

Comment:
It would be better to implment J^theta_PEST with that formula as well to avoid building radial derivatives of stream function